cosmos/constants.py

```diff
@@ -31,14 +29,38 @@
 PARTIALLY_SUPPORTED_AIRFLOW_VERSIONS = [Version("2.9.0"), Version("2.9.1")]
 
 
-S3_FILE_SCHEME = "s3"
-GS_FILE_SCHEME = "gs"
 ABFS_FILE_SCHEME = "abfs"
 
-FILE_SCHEME_AIRFLOW_DEFAULT_CONN_ID_MAP = {
-    S3_FILE_SCHEME: S3Hook.default_conn_name,
-    GS_FILE_SCHEME: GCSHook.default_conn_name,
-    ABFS_FILE_SCHEME: WasbHook.default_conn_name,
+
+def _default_s3_conn() -> str:
+    from airflow.providers.amazon.aws.hooks.s3 import S3Hook
+
+    return S3Hook.default_conn_name
+
+
+def _default_gcs_conn() -> str:
+    from airflow.providers.google.cloud.hooks.gcs import GCSHook
+
+    return GCSHook.default_conn_name
+
+
+def _default_wasb_conn() -> str:
```

Comment:
Did specifying a `wasb` URI work for you? Can you please share the format of the `wasb` URI you used? Perhaps we could include that URI format in our example DAG & docs. 
It didn't seem to work when I tried it, which is why I didn't include `wasb`—I considered it a limitation that we only support `abfs`. I tested various schemes like `abfs`, `wasb`, `azure`, and `az`, and was only able to successfully run our `cosmos_manifest_example.py` DAG with the `abfs` scheme, using the URI format specified in [this line](https://github.com/astronomer/astronomer-cosmos/blob/7889fbe7abf3ad79a14530e91ed90ef6352eba4e/dev/dags/cosmos_manifest_example.py#L93).

Reply:
Yes, I see what you mean, it seems that `wasb://` is not supported: https://github.com/fsspec/universal_pathlib/blob/92675f060775e7aeb3ffc92cb74abdb92458d231/README.md?plain=1#L86 It also seems that:

- In the `upath` library, `az:`, `adl:`, `abfs:` and `abfss:` are also all valid schemes, albeit Airflow only references the latter 3 in the code: https://github.com/apache/airflow/blob/e3824eaaba7eada9a807f7a2f9f89d977a210e15/airflow/providers/microsoft/azure/fs/adls.py#L29 So I think we should add `adl:`, `abfs:` and `abfss:`.
- Upon review, I'm actually really confused by the difference between the "data lake" hook and the "wasb" hook. 🤔 Azure is confusing. I am going to revert back to using the `WasbHook` default. I have a feeling that they both take in the same configuration but that the WasbHook is more commonly used.
- Seems like we may want to do a few updates to how the Cosmos dbt docs plugin works with Azure blob storage, too, but that is a separate PR.

Thanks for the review!